enrichments/trace/internal/elastic/span.go

```diff
@@ -200,6 +201,10 @@ func (s *spanEnrichmentContext) Enrich(
 			s.userAgentName = v.Str()
 		case semconv27.AttributeUserAgentVersion:
 			s.userAgentVersion = v.Str()
+		case "type":
```

Comment:
How will this attribute be set?

Reply:
It's set by the EDOT mobile SDKs [specified here](https://github.com/elastic/apm/blob/main/specs/agents/mobile/README.md#additional-span-attributes).

Reply:
Maybe also add `elasticattr.TransactionType` do this case so that if `transaction.type` is set already, it's not overridden?

Reply:
> Maybe also add `elasticattr.TransactionType` do this case so that if `transaction.type` is set already, it's not overridden?

I'm ok with validating that `transaction.type` won't change if it's already present. It's a different use case though it seems like it'd be easy enough to add it in this PR. What do you think, @gregkalapos @lahsivjar ?

Reply:
> Maybe also add elasticattr.TransactionType do this case so that if transaction.type is set already, it's not overridden?

I think overriding behaviour should be handled for other attributes too, but I am not sure if not overriding in all cases would be the correct option. I have created https://github.com/elastic/opentelemetry-lib/issues/185 to discuss it further.

Reply:
Ok by me to include it in this PR.

Reply:
> I think overriding behaviour should be handled for other attributes too, but I am not sure if not overriding in all cases would be the correct option. I have created https://github.com/elastic/opentelemetry-lib/issues/185 to discuss it further.

Got it. Thanks, @lahsivjar. I won't add that change here then.

Reply:
I think we should do it in a separate PR since this concern might apply to other attributes too, but I don't have a strong opinion here. Does that makes sense @gregkalapos @felixbarny ? Or do we think only `transaction.type` needs to be handled this way?